modules/affile/affile-source.c

```diff
@@ -253,6 +253,43 @@ affile_sd_construct_proto(AFFileSourceDriver *self, gint fd)
     }
 }
 
+static void
+affile_sd_reset_file_state(PersistState *state, const gchar *old_persist_name)
+{
+  gchar *new_persist_name = g_strdup_printf("%s_DELETED",old_persist_name);
+  persist_state_rename_entry(state,old_persist_name,new_persist_name);
+  g_free(new_persist_name);
+}
+
+static inline gboolean
+_setting_logpipe(LogPipe *s)
+{
+  AFFileSourceDriver *self = (AFFileSourceDriver *) s;
+
+  log_pipe_append((LogPipe *) self->reader, s);
+
+  if (!log_pipe_init((LogPipe *) self->reader))
+    {
+      log_pipe_unref((LogPipe *) self->reader);
+      return FALSE;
+    }
+  return TRUE;
+}
```

Comment:
I am not sure if it's a good idea to merge the append and the init operation in this function. I understand that the same sequence was used on the call site, but I have trouble giving this function a good name.

_prepend_and_init_log_reader()

but that clearly violates SRP. So, for now I don't like this extraction, it just makes the code more difficult to understand.